src/platform_impl/macos/window_delegate.rs

```diff
@@ -419,32 +421,62 @@ declare_class!(
         }
     }
 
+    // Key-Value Observing
     unsafe impl WindowDelegate {
-        // Observe theme change
-        #[method(effectiveAppearanceDidChange:)]
-        fn effective_appearance_did_change(&self, sender: Option<&AnyObject>) {
-            trace_scope!("effectiveAppearanceDidChange:");
-            unsafe {
-                self.performSelectorOnMainThread_withObject_waitUntilDone(
-                    sel!(effectiveAppearanceDidChangedOnMainThread:),
-                    sender,
-                    false,
-                )
-            };
-        }
+        #[method(observeValueForKeyPath:ofObject:change:context:)]
+        fn observe_value(
+            &self,
+            key_path: Option<&NSString>,
+            _object: Option<&AnyObject>,
+            change: Option<&NSDictionary<NSKeyValueChangeKey, AnyObject>>,
+            _context: *mut c_void,
+        ) {
+            trace_scope!("observeValueForKeyPath:ofObject:change:context:");
+            if key_path == Some(ns_string!("effectiveAppearance")) {
```

Comment:
Given that entire function is inside the `if` probably should just early return.

Reply:
At some point in the future we might get more `key_path`s that we're observing here, and then it makes sense to handle each in their own `if` - that's the whole reason why I'm checking the keypath, in reality we don't have to right now, since we're only observing one.

Reply:
In the future, I'd prefer each if case like that to be in its own _function/method_, but for now, I'd use early return, because it's the best thing you can do _now_.